core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -2111,24 +2111,45 @@ public IRubyObject index(ThreadContext context, Block block) {
 
     @JRubyMethod
     public IRubyObject bsearch(ThreadContext context, Block block) {
-        Ruby runtime = context.runtime;
-
         if (!block.isGiven()) {
             return enumeratorize(context.runtime, this, "bsearch");
         }
 
+        int rVal = bsearch_index_internal(context, block);
+        if(rVal == -1) {
```

Comment:
Not a big deal, but I'm curious why the two `if(rVal...` statements lack a space between the `if` and `(`, but the other `if` statements do not.